server/chroma/src/server.rs

```diff
@@ -0,0 +1,112 @@
+use std::time::Duration;
+
+use actix_cors::Cors;
+use actix_web::{web, App, HttpServer};
+use anyhow::{anyhow, bail, Result};
+use cabbage::KoalaApi;
+use noiseless_tracing_actix_web::NoiselessRootSpanBuilder;
+use tracing::{info, warn};
+use tracing_actix_web::TracingLogger;
+
+use dal::database::Database;
+use dal::storage_engine::{S3Config, Storage};
+
+use crate::config::Config;
+use crate::routes;
+use crate::routes::appdata::{AlbumIdCache, AppData, Ratelimits, SessionIdCache, WebData};
+use crate::routes::routable::Routable;
+
+/// Run the chroma server and will block until the server is stopped or crashes
+///
+/// # Errors
+///
+/// If the configuration is invalid
+/// If the database connection cannot be completed
+/// If the S3 configuration is invalid
+/// If the port for the server is already in use
+/// If a problem occurs in one of the server routes
+pub async fn run() -> Result<()> {
+    // Initialize chroma's core components
+    let config = init_config()?;
+    let db = init_database(&config).await?;
+    let storage = init_storage(&config).await?;
+
+    // Package the core components up into the AppData struct
+    let app_data = AppData {
+        koala: KoalaApi::new(config.koala_base_redirect_uri().clone())?,
+        db,
+        storage,
+        config,
+        ratelimits: Ratelimits::new(),
+    };
+
+    // Run the webserver using the AppData until stopped or crash
+    start_webserver(app_data).await
+}
+
+/// Parses the configuration using
+fn init_config() -> Result<Config> {
```

Comment:
Does this belong in the server module? feels like a thing that should be in main

Reply:
Logically yes, but I felt that main.rs got a bit to cluttered if all the initialization steps were done there. That being said, I'm also not entirely happy with how drunk me did it now, so I will see if I can cook something up